### ISSUE TITLE
Availability of auto-suggestions not announced to screen readers 

### ISSUE THREAD

**Issue by kaylajheffernan** [moroshko/react-autosuggest#279]
The ‘Where’ field displays a list of location suggestions as you type. The suggestions are keyboard accessible using the down arrow key, however the fact that there are auto-suggest options available is not announced to screen readers, so unless the user explores, they won’t know the options exist. 

**Recommendation** 

Announce to screen readers that location suggestions are available. 

**Option 1:** Add hidden text as part of the `<label>` mentioning that suggestions will appear below the field. 
```html
<label>
  Where
  <span class=”sr-only”>
    Suggestions will appear below the field as you type, use down arrow to navigate to the list
  </span>
</label> 
```

**Option 2:** Add an `aria-live` region to announce the availability of the options when the list of options is displayed. Insert the message into the `aria-live` region when auto-suggest results first appear. 
```html
<div aria-live=”polite”>
  Location suggestions are available, use the down arrow key to navigate to the list
</div>
```

**Comment by moroshko** [comment]
@kaylajheffernan **Option 1** is totally in your control. The `<label>` is not part of `react-autosuggest`.

I had a quick look at the docs of [`aria-live`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions), and it feels not mature yet.

Do you know how some of the well known sites (e.g. Google) solve this problem?

Also, [this accessibility issue](https://github.com/moroshko/react-autosuggest/issues/273) seems related. I'd appreciate your feedback on it.

**Comment by kaylajheffernan** [comment]
Thanks @moroshko . We'll go with Option 1 then for now. 

When I've played with Google & voice over it doesn't announce the auto-suggestions. I guess the difference is that you can enter free text, whereas we have structured data so we really need to be announcing to them the results are there. 

Not sure about announcing the number of options in the dropdown. Would it interrupt them as they are typing and the number dynamically changes as matches change?

**Comment by moroshko** [comment]
@kaylajheffernan Good point regarding announcing the number of suggestions. I wonder if there is an official guideline about this.

I'll close this issue for now since you have a workable solution.

**Comment by LarsHassler** [comment]
I've done some research, because i have to write an accessible autocomplete in vanilla javascript for a client and found that the [jQuery UI](https://jqueryui.com/autocomplete/) did this by using an status container:

```html
<div role="status" aria-live="assertive" aria-relevant="additions" class="ui-helper-hidden-accessible">
  <div style="display: none;">No search results.</div>
  <div>10 results are available, use up and down arrow keys to navigate.</div>
</div>
```

But unfortunately I could not find any best practices guides around this and most of the autocompletes i could find did not announced the number of results.
Therefore right now i'm not using it.

P.S.:
But some screen readers announce the position of the item if the `<li>`s are wrapped inside of an ordered list, which i'm going for.

**Comment by rheydt** [comment]
Is this still on the roadmap for future improvements?
